rfcs/20200705-tfx-ir.md

```diff
@@ -0,0 +1,1563 @@
+    zero or more `Artifacts` and produces zero or more `Artifacts`. The
+    non-Artifact inputs (e.g., parameters) of the execution will be stored as
+    the properties of the `Execution` instance.
+*   **Context**: a `Context` describes grouping concepts of workflow elements.
+    In TFX, it does not map to any specific entity, instead it groups artifacts
+    and executions under certain meanings (e.g., a pipeline run). Note that an
+    `Execution` or an `Artifact` can be linked to more than one `Contexts`.
+
+There are also three types of edges that are used in TFX, linking the
+aforementioned vertices:
+
+*   **Event**: an `Event` links an `Artifact` and an `Execution`. In TFX, the
+    `Event` contains the following info:
+    *   The type of the `Event`. For example, type `INPUT` means that the
+        `Artifact` is an input of the `Execution`.
+    *   The key of the input / output `Artifact`. For example, a `Trainer` might
+        produce two groups of models: one group of models are normal TensorFlow
+        models and the other group of models are optimized for mobile. The key
+        can be used to differentiate them.
+    *   The index of the input / output `Artifact`. For example, an `ExampleGen`
+        might produce a list of outputs with the same output key. To
+        differentiate them, the index will be used.
+*   **Attribution**: an `Attribution` links an `Artifact` to a `Context`.
+*   **Association**: an `Association` links an `Execution` to a `Context`.
+
+#### 3.1.2. Data model of a Channel
+
+In TFX DSL, artifacts flow between components in
+[`Channels`](https://www.tensorflow.org/tfx/api_docs/python/tfx/types/Channel).
+In the simple example below, `MyTrainer` uses the output `Channel` of
+`MyExampleGen` as its input `Channel` for examples data. In other words,
+`MyExampleGen` is writing to its output `Channel` and `MyTrainer` is reading
+from the same `Channel`.
+
+```python
+def create_pipeline():
+  eg = MyExampleGen(param_one=...)
+  tr = MyTrainer(input_examples=eg.outputs['output_examples'], ...)
+
+  return Pipeline.pipeline(
+      pipeline_name='my_pipeline',
+      components=[eg, tr],
+      execution_mode=SYNC
+      ...)
+```
+
+<div align="center"><em><a id="table1">Table 1</a>: A simple pipeline.</em></div>
+
+`Channel` is an abstract concept in TFX instead of a real entity in MLMD like
+`Artifact`. `Channel`. It is built on top of the basic elements described
+previously. When being mapped to MLMD, a `Channel` is essentially a view of a
+collection of `Artifact` that satisfies a set of restrictions defined by five
+categories of predicates shown below. Note that the predicates soly rely on
+information from MLMD and `ArtifactType` is the only required one.
+
+Required? | Type of Predicates
+:-------: | :------------------------------------------------------------------
+Yes       | Predicates on the type of artifacts
```

Comment:
Sorry I'm new to TFX Channel, what are the supported `type` of artifacts in Channel? Is it possible to define the artifact with NoType?

Reply:
The current supported 'type' can be any MLMD artifacttype.
A `GenericType` is something we're working on.

/cc @charlesccychen 
/cc @zhitaoli